src/views/ingresses/actions/hooks/useIngressActions.ts

```diff
@@ -0,0 +1,58 @@
+import { useNavigate } from 'react-router-dom-v5-compat';
+
+import { IngressModel, modelToRef } from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiNetworkingV1Ingress } from '@kubevirt-ui/kubevirt-api/kubernetes/models';
+import {
+  Action,
+  useAnnotationsModal,
+  useDeleteModal,
+  useLabelsModal,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { asAccessReview, getName, getNamespace } from '@utils/resources/shared';
+
+type UseIngressActions = (ingress: IoK8sApiNetworkingV1Ingress) => [actions: Action[]];
+
+const useIngressActions: UseIngressActions = (ingress) => {
+  const { t } = useNetworkingTranslation();
+
+  const navigate = useNavigate();
+  const launchDeleteModal = useDeleteModal(ingress);
+  const launchLabelsModal = useLabelsModal(ingress);
+  const launchAnnotationsModal = useAnnotationsModal(ingress);
+
+  const ingressNamespace = getNamespace(ingress);
+  const ingressName = getName(ingress);
+
+  const actions = [
+    {
+      accessReview: asAccessReview(IngressModel, ingress, 'update'),
+      cta: launchLabelsModal,
+      id: 'edit-labels-ingresses',
+      label: t('Edit labels'),
+    },
+    {
+      accessReview: asAccessReview(IngressModel, ingress, 'update'),
+      cta: launchAnnotationsModal,
+      id: 'edit-annotations-ingresses',
+      label: t('Edit annotations'),
+    },
+    {
+      accessReview: asAccessReview(IngressModel, ingress, 'update'),
+      cta: () =>
+        navigate(`/k8s/ns/${ingressNamespace}/${modelToRef(IngressModel)}/${ingressName}/yaml`),
```

Comment:
Can you check getK8sResourcePath function from sdk to see if it works for you?
It would be nicer than building the URL yourself :)

Reply:
Done!